changelog/@unreleased/pr-5146.v2.yml

```diff
@@ -0,0 +1,9 @@
+type: fix
+fix:
+  description: Calling registerWatches now tells the lockWatchManager to _only_ watch
+    those watches, instead of adding it to the set of all registered watches so far.
+    Note that this does not affect Timelock (which will still contain all watches
+    registered so far), but rolling Timelock will mean that the LockWatchManager now
+    only re-registers the latest requested watches.
```

Comment:
I'd probably re-phrase this in terms of what the client sees (i.e. the impl details are not needed, I think the last part is the key here - `The LockWatchManager now only re-registers the latest set of requested watches when TimeLock is bounced`).